### PR TITLE
add in-app and system notifications 

### LINKED ISSUE #527: Implement toasts notifications on pages in the page package
Some pages have been moved to the page package and don't have access to the notification implementation made on the common struct. Notification implementation should be added for pages in the page package.

### PR DESCRIPTION
**Pull request by oshorefueled**
This PR resolves #527 

It implements and in-app notification (toast) which is used in displaying success or error messages to a user within the app in response to an action. It also implements a system notification which displays os specific system-wide applications. Some of the notifications displayed with the system notification could be new block notifications, receive transaction notifications, transaction confirmation notification e.t.c..

Calls to the CreateToast method used by pages in the page package are replaced with the Notify method created in this PR.

### CODE REVIEW

File: ui/notification/toast.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	return &Toast{
+		theme: th,
+	}
+}
+
+func (t *Toast) Notify(message string, success bool) {
+	t.message = message
+	t.success = success
+	t.timer = time.NewTimer(time.Second * 3)
+	t.show = true
```
Comment (beansgum) [inline ui/notification/toast.go]: Protect these variables with a mutex.

File: ui/notification/toast.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+func (t *Toast) Layout(gtx layout.Context) layout.Dimensions {
+	if !t.show {
```
Comment (beansgum) [inline ui/notification/toast.go]: Remove the show variable and replace this condition with `t.timer == nil`.

File: ui/notification/toast.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		return layout.Dimensions{}
+	}
+
+	t.handleToastDisplay(gtx)
```
Comment (beansgum) [inline ui/notification/toast.go]: This check should be done at the top of the function.

File: ui/notification/log.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017, The dcrdata developers
```
Comment (beansgum) [inline ui/notification/log.go]: This package does not need its own logger.

File: ui/notification/system.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+	return &SystemNotification{
+		iconPath: filepath.Join(absolutePath, icon),
+	}
+}
+
+func (s *SystemNotification) Notify(message string) {
+	err := beeep.Notify(title, message, s.iconPath)
+	if err != nil {
+		log.Info("could not initiate desktop notification, reason:", err.Error())
```
Comment (beansgum) [inline ui/notification/system.go]: This error should be returned not logged.

File: ui/notification/utils.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package notification
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func getAbsolutePath() (string, error) {
```
Comment (beansgum) [inline ui/notification/utils.go]: I don't think any utils.go file is needed in this package since it's just this function and it's used only in system.go

**Review by beansgum — CHANGES_REQUESTED** [review]

File: ui/notification/system.go (hunk 5 above)
Comment (oshorefueled) [reply ui/notification/system.go]: The log avoids handling notification errors in the ui code. All that needs to be called is the Notify method, just like toast notifications.

File: ui/notification/toast.go (hunk 3 above)
Comment (oshorefueled) [reply ui/notification/toast.go]: moving the handle function to the top of the Layout method would lead to a nil dereference error.

**Review by beansgum — APPROVED** [review]

**Review by Sirmorrison — APPROVED** [review]
Tested and looks good.

**Pull request merged**